app/views/events/_event.html.erb

```diff
@@ -28,7 +28,9 @@
       </div>
 
       <div class="pull-right">
-        <%= render :partial => 'events/apply_button', :locals => {event: event} %>
+        <% if can? :apply, Event %>
+            <%= render :partial => 'events/apply_button', :locals => {event: event} %>
```

Comment:
To reduce code duplication you can put this check inside the apply_button partial. I think currently you even forgot one position to add this test.